packages/blade/src/components/Input/BaseInput/__tests__/BaseInput.web.test.tsx

```diff
@@ -0,0 +1,177 @@
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" validationState="success" successText="Success" />,
+    );
+
+    const input = getByLabelText(label);
+    const successText = getByText('Success');
+
+    expect(successText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Success');
+    expect(input).toBeValid();
+  });
+
+  it('should display error validation state', () => {
+    const label = 'Enter name';
+    const { getByText, getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" validationState="error" errorText="Error" />,
+    );
+
+    const input = getByLabelText(label);
+    const errorText = getByText('Error');
+
+    expect(errorText).toBeTruthy();
+    expect(input).toHaveAccessibleDescription('Error');
+    expect(input).toBeInvalid();
+  });
+
+  it('should render with icons', () => {
+    const { container } = renderWithTheme(
+      <BaseInput
+        label="Enter name"
+        placeholder="First Last"
+        id="name"
+        leadingIcon={EyeIcon}
+        trailingIcon={CloseIcon}
+      />,
+    );
+
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should be focussed when autoFocus flag is passed', () => {
+    const label = 'Enter name';
+    // eslint-disable-next-line jsx-a11y/no-autofocus
+    const { getByLabelText } = renderWithTheme(<BaseInput label={label} id="name" autoFocus />);
+
+    const input = getByLabelText(label);
+    expect(input).toHaveFocus();
+  });
+
+  it('should be disabled when isDisabled flag is passed', () => {
+    const label = 'Enter name';
+    const { getByLabelText } = renderWithTheme(<BaseInput label={label} id="name" isDisabled />);
+
+    const input = getByLabelText(label);
+    expect(input).toBeDisabled();
+  });
+
+  it('should handle onChange', async () => {
+    const label = 'Enter name';
+    const onChange = jest.fn();
+    const user = userEvent.setup();
+    const userName = 'Divyanshu';
+
+    const { getByLabelText } = renderWithTheme(
+      <BaseInput label={label} id="name" name="name" onChange={onChange} />,
+    );
+
+    const input = getByLabelText(label);
+    await user.type(input, userName);
+
+    // should be called for each keystroke
+    expect(onChange).toHaveBeenCalledTimes(userName.length);
+    expect(onChange).toHaveBeenLastCalledWith({ name: 'name', value: userName });
+  });
+
+  it('should handle onBlur', async () => {
+    const user = userEvent.setup();
+    const label = 'Enter name';
+    const userName = 'Divyanshu';
+    const onBlur = jest.fn();
+
+    renderWithTheme(
+      <BaseInput
+        label={label}
+        id="name"
+        name="name"
+        defaultValue={userName}
+        // eslint-disable-next-line jsx-a11y/no-autofocus
+        autoFocus
+        onBlur={onBlur}
+      />,
+    );
+
+    // shifts user focus and therefore blurs the focussed input
+    await user.tab();
```

Comment:
This might seem unnecessary but it's better to remove autofocus and use the input how the user might use it. first focus it then blur it manually. 

Reason being autofocus even works when the component is set to `tabIndex={-1}` which removes the elements taking focus from keyboard input but still accepts programatic focus (e.focus/autofocus attr)

```ts
    // focus
    await user.tab();
    // shifts user focus and therefore blurs the focussed input
    // blur
    await user.tab();
```

Reply:
I think this would be unrelated to current test (since it's for `onBlur`), though maybe we can add another test for what you're mentioning.

Reply:
Added another case in https://github.com/razorpay/blade/pull/695/commits/8be6b8367acaeaf5f157a5ba38eb97ff0c75f785